### DISCUSSION TITLE
Spotlight range: I need help to set it dynamically to lighten different distances

### DISCUSSION THREAD

**Discussion by ThomasCartier** [bevyengine/bevy#14475] [Q&A] [answered]
Hi,
I currently have a massive issue to properly set my spotlights to lighten my scenes, from different positions, and directions toward a target.

```
commands.spawn(SpotLightBundle {
        transform: Transform::from_xyz(5., 10.0, 0.).looking_at(Vec3::new(5.0, 0.0, 0.), Vec3::Y),
        spot_light: SpotLight {
                intensity: 400_000.0, // lumens
                color: Color::WHITE,
                shadows_enabled: true,
                inner_angle: 0.,
                outer_angle: PI / 2.,
                range: 500.,
                ..Default::default()
        },
        ..default()
});
```

I have hard time deciding the proper values for the intensity, range. 
What is the formula for a spotlight to lighten at a certain radius on the ground with a certain intensity on this ground, all that depending on the distance from the target point it is located ?

Thanks

**Comment by bugsweeper** [comment, marked as answer]
Short answer: According to the laws of physics (and realistic rendering should be very close to the laws of physics), the total brightness from a light source at any distance is distributed evenly over the sphere regardless of the radius of the sphere (distance). As the radius increases, the area of ​​the sphere increases quadratically, and the brightness decreases proportionally. For example, if the radius/distance is increased by 2 times, the brightness will decrease by 4 times.
You can read in more detail in the article: [Physically Based Rendering in Filament](https://google.github.io/filament/Filament.html#mjx-eqn-pointLightLuminousPower) at chapter 5.2.2.2 Spot lights

**Reply by ThomasCartier** [reply]
Thanks, I will code that
